libraries/botbuilder-ai/tests/qnaMakerDialog.test.js

```diff
@@ -1,12 +1,55 @@
 const { QnAMakerDialog, QnAMaker } = require('../lib');
-const { DialogSet } = require('botbuilder-dialogs');
+const { Dialog, DialogSet, DialogTurnStatus, DialogManager } = require('botbuilder-dialogs');
 const { ok, strictEqual } = require('assert');
+const { ConversationState, MemoryStorage, TestAdapter } = require('botbuilder-core');
+const fs = require('fs');
+const nock = require('nock');
+
+const path = require('path');
+const ENV_FILE = path.join(__dirname, '.env');
+require('dotenv').config({ path: ENV_FILE });
+
+const KB_ID = process.env.QNAKNOWLEDGEBASEID;
+const ENDPOINT_KEY = process.env.QNAENDPOINTKEY;
+const HOSTNAME = process.env.QNAHOSTNAME || 'test-qna-app';
+const isMockQna = false || !(KB_ID && ENDPOINT_KEY);
+
+const beginMessage = { text: `begin`, type: 'message' };
+
 
-const KB_ID = 'kbId';
-const ENDPOINT_KEY = 'endpointKey';
 
 describe('QnAMakerDialog', function() {
-    this.timeout(3000);
+    // this.timeout(3000);
+    this.timeout(120000);
+    const testFiles = fs.readdirSync(`${ __dirname }/TestData/${ this.title }/`);
+
+    beforeEach(function(done){
+        nock.cleanAll();
+        if (isMockQna) {
+            var fileName = replaceCharacters(this.currentTest.title);
+            var filePath = `${ __dirname }/TestData/${ this.test.parent.title }/`;
+            var arr = testFiles.filter(function(file) { return file.startsWith(fileName + '.')} );
+    
+            arr.forEach(file => {
+                nock(`https://${ HOSTNAME }.azurewebsites.net`)
+                    .matchHeader('User-Agent', /botbuilder-ai\/4.*/)
+                    .post(/qnamaker/)
+                    .replyWithFile(200, filePath + file);
+            });
+        }
+        done();
+    })
+    
+    afterEach(function(done){
+        nock.cleanAll();
+        done();
+    });
+    
+    function replaceCharacters (testName, testDesc) {
+        return testName
```

Comment:
Prettier should format this slightly differently.

Reply:
Ran prettier 🌷